src/include/txn.i

```diff
@@ -885,9 +889,13 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_ITEM *key, uint
         return (0);
     }
 
-    /* If the start time pair is visible then we need to return the ondisk value. */
-    if (__wt_txn_visible(session, start.txnid, start.timestamp) ||
-      F_ISSET(session, WT_SESSION_RESOLVING_MODIFY)) {
+    /*
+     * If the start time pair is visible and it is not a prepared value then we need to return the
+     * ondisk value.
+     */
+    if ((!prepare || (stop.txnid != WT_TXN_MAX && stop.timestamp != WT_TS_MAX)) &&
```

Comment:
we need to return WT_PREPARE_CONFLICT if there is a prepared update on disk. I am not seeing this case here.

Reply:
`WT_PREPARE_CONFLICT` is returned only for the transaction that is not configured `ignore_prepare` flag.  The prepared update is already present as part of the update chain whenever the page gets loaded into memory, so the error should have already raised when it is processing the update chain. When the code flow is accessing the on-disk data to verify means that the transaction is already configured with `ignore_prepare` so we should just make the on-disk data as not visible and proceed to the history store instead of adding a check to return `WT_PREPARE_CONFLICT`?